src/main/java/de/komoot/photon/searcher/AbstractPhotonRequestHandler.java

```diff
@@ -0,0 +1,40 @@
+package de.komoot.photon.searcher;
+
+import de.komoot.photon.query.PhotonRequest;
+import de.komoot.photon.query.TagFilterQueryBuilder;
+import de.komoot.photon.utils.ConvertToJson;
+import org.elasticsearch.action.search.SearchResponse;
+import org.json.JSONObject;
+
+import java.util.List;
+
+/**
+ * Created by Sachin Dole on 2/20/2015.
+ */
+public abstract class AbstractPhotonRequestHandler<R extends PhotonRequest> implements PhotonRequestHandler<R> {
+
+    private ElasticsearchSearcher elasticsearchSearcher = new BaseElasticsearchSearcher();
+
+    @Override
+    public final List<JSONObject> handle(R photonRequest) {
+        TagFilterQueryBuilder queryBuilder = buildQuery(photonRequest);
+        Integer limit = photonRequest.getLimit();
+        SearchResponse results = elasticsearchSearcher.search(queryBuilder.buildQuery(), limit);
+        if (results.getHits().getTotalHits() == 0) {
+            results = elasticsearchSearcher.search(queryBuilder.withStrictMatch().buildQuery(), limit);
```

Comment:
shouldn't it be the other way around? first a strict match query. if there is no match, make a less strict query.